TodoApp/tests/TodoApp.Services.Tests/Wrappers/DateTimeProviderTests.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Threading;
+using NUnit.Framework;
+using TodoApp.Contracts.Wrappers;
+using TodoApp.Services.Wrappers;
+
+namespace TodoApp.Services.Tests.Wrappers
+{
+    [TestFixture]
+    internal class DateTimeProviderTests
+    {
+        private IDateTimeProvider _dateTimeProvider;
+
+        [SetUp]
+        public void Init()
+        {
+            _dateTimeProvider = new DateTimeProvider();
+        }
+
+        [Test]
+        public void ReturnsDifferentDateTimes_OnMultipleCalls()
+        {
+            var time1 = _dateTimeProvider.GetCurrentDateTime();
+            Thread.Sleep(1000);
```

Comment:
This is a bit too much IMO. Second is next to eternity in an application lifetime :)
Read a bit more on [Thread.Sleep](https://msdn.microsoft.com/fi-fi/library/d00bd51t(v=vs.110).aspx) (especially `remarks`). While it is not completely correct approach, I believe that relatively small number (tents) of milliseconds should suffice. (of course, you can choose the more-or-less correct way to approach waiting as described in this [SO](https://stackoverflow.com/questions/6254703/thread-sleep-for-less-than-1-millisecond))